source/MaterialXCore/Document.cpp

```diff
@@ -782,12 +830,170 @@ void Document::upgradeVersion()
+                    }
+                }
+
+                modified = true;
+
+                // Find the shader type if defined
+                string shaderNodeType = SURFACE_SHADER_TYPE_STRING;
+                NodeDefPtr nodeDef = sr->getNodeDef();
+                if (nodeDef)
+                {
+                    shaderNodeType = nodeDef->getType();
+                }
+
+                // Add in a new shader node
+                const string shaderNodeCategory = sr->getNodeString();
+                NodePtr shaderNode = addNode(shaderNodeCategory, shaderNodeName, shaderNodeType);
+                shaderNode->setSourceUri(sr->getSourceUri());
+
+                for (auto valueElement : sr->getChildrenOfType<ValueElement>())
+                {
+                    ElementPtr portChild = nullptr;
+
+                    // Copy over bindinputs as inputs, and bindparams as params
+                    if (valueElement->isA<BindInput>())
+                    {
+                        portChild = shaderNode->addInput(valueElement->getName(), valueElement->getType());
+                    }
+                    else if (valueElement->isA<BindParam>())
+                    {
+                        portChild = shaderNode->addParameter(valueElement->getName(), valueElement->getType());
+                    }
+                    if (portChild)
+                    {
+                        // Copy over attributes.
+                        // Note: We preserve inputs which have nodegraph connections,
+                        // as well as top level output connections.
+                        portChild->copyContentFrom(valueElement);
+                    }
+                }
+
+                // Copy over any bindtokens as tokens
+                for (auto bt : sr->getBindTokens())
+                {
+                    TokenPtr token = shaderNode->addToken(bt->getName());
+                    token->copyContentFrom(bt);
+                }
+
+                // Create a new material node if not already created and
+                // add a reference from the material node to the new shader node
+                if (!materialNode)
+                {
+                    // Set the type of material based on current assumption that
+                    // surfaceshaders + displacementshaders result in a surfacematerial
+                    // while a volumeshader means a volumematerial needs to be created.
+                    string materialNodeCategory =
+                        (shaderNodeType != VOLUME_SHADER_TYPE_STRING) ? SURFACE_MATERIAL_NODE_STRING
+                        : VOLUME_MATERIAL_NODE_STRING;
+                    materialNode = addNode(materialNodeCategory, materialName, MATERIAL_TYPE_STRING);
+                    materialNode->setSourceUri(m->getSourceUri());
+                    // Note: Inheritance does not get transfered to the node.
+                    //materialNode->setInheritString(m->getInheritString()); 
```

Comment:
Did you mean to commit this commented out line?

Reply:
No. I'll remove it.